crates/biome_grit_patterns/src/grit_analysis_ext.rs

```diff
@@ -0,0 +1,69 @@
+use biome_diagnostics::{Diagnostic, PrintDescription, Severity};
+use grit_util::{AnalysisLog, Position, Range};
+use std::path::{Path, PathBuf};
+
+pub trait GritAnalysisExt {
+    fn to_log(&self, path: Option<&Path>) -> AnalysisLog;
+}
+
+impl<T> GritAnalysisExt for T
+where
+    T: Diagnostic,
+{
+    fn to_log(&self, path: Option<&Path>) -> AnalysisLog {
+        let location = self.location();
+        let source = location.source_code;
+        let range =
+            match (location.span, source) {
+                (Some(range), Some(source)) => source.text[..range.start().into()]
+                    .lines()
+                    .enumerate()
+                    .last()
+                    .map(|(i, line)| {
+                        let start = Position {
+                            line: (i + 1) as u32,
+                            column: line.len() as u32,
+                        };
+                        let end = source.text[range].lines().enumerate().last().map_or(
+                            start,
+                            |(j, line)| Position {
+                                line: start.line + j as u32,
+                                column: if j == 0 {
+                                    start.column + line.len() as u32
+                                } else {
+                                    line.len() as u32
+                                },
+                            },
+                        );
+                        Range {
+                            start,
+                            end,
+                            start_byte: range.start().into(),
+                            end_byte: range.end().into(),
+                        }
```

Comment:
I am not sure about the intention of this code, but we have a utility to extract columns and rows from the source code, I suggest to look here and see how we use it: 

https://github.com/biomejs/biome/blob/cd6e830b48d06939ca42f580dd31a858846f0884/crates/biome_diagnostics/src/display_github.rs#L30-L32